storage/redis_adapter.go

```diff
@@ -54,12 +54,9 @@ func (mp messagePipeliner) execPipeline() {
 			mp.errCh <- fmt.Errorf("Error adding ltrim of %s to the pipeline: %s", app, err)
 		}
 	}
-	go func() {
-		defer mp.pipeline.Close()
-		if _, err := mp.pipeline.Exec(); err != nil {
-			mp.errCh <- fmt.Errorf("Error executing pipeline: %s", err)
-		}
-	}()
+	if _, err := mp.pipeline.Exec(); err != nil {
+		mp.errCh <- fmt.Errorf("Error executing pipeline: %s", err)
+	}
```

Comment:
This was deliberately being executed as a goroutine-- because there's no reason for this code to block on pipeline execution.